transcom/mymove
# B 21580 unlock gov search queues int

## [B-21580](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1024956)

## Summary

Previously when a QAE, GSR, CSR, or Headquarters office user would search/open a move, and then click the MilMove button and search for the move again they would see that the move was locked.

This PR adds the same unlocking capability that the SC/TIO/TOO has for the search queue for those roles.

## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

1. Before running make/client servers change `FEATURE_FLAG_MOVE_LOCK=true` in your envrc
2. Run `direnv allow` and then you can start up your make/client servers
3. Login as QAE, GSR, CSR, or Headquarters office user
4. Search for any existing move like DRTTFR
5. Run sql in db to confirm move is unlocked
6. Click record for move to see move details page
7. Run **sql** in db to confirm move is locked
8. Click Milmove button or naviagte away from move(signing out, closing out the page)
9. Run **sql** in db to confirm move is unlocked again
10. Search for the same move you opened previously
11. Confirm there is no lock next to the record move code

**sql example:** `select id, locked_by, lock_expires_at  from moves m where "locator" = 'DRTTFR'`
if lockedby/lock_expires_at is null there is no lock on that move 
### Backend


## Screenshots
**_Before_**
<img width="1682" alt="Screenshot 2024-11-04 at 11 53 14 PM" src="https://github.com/user-attachments/assets/648868a7-7510-42b3-817b-3bcb554db9e6">


**_After_**
<img width="1678" alt="Screenshot 2024-11-04 at 11 43 40 PM" src="https://github.com/user-attachments/assets/55ca3819-f354-496a-8573-2abcb4a20682">
